addons/account/models/account_move.py

```diff
@@ -1408,33 +1408,75 @@ def create_exchange_rate_entry(self, aml_to_fix, amount_diff, diff_in_currency,
             move.post()
         return line_to_reconcile, partial_rec
 
-    @api.model
-    def create(self, vals):
-        res = super(AccountPartialReconcile, self).create(vals)
+    def _fix_multiple_exchange_rates_diff(self, amls_to_fix, amount_diff, diff_in_currency, currency, move):
+        self.ensure_one()
+        MoveLineObj = self.env['account.move.line'].with_context(check_move_validity=False)
+        PartialRecObj = self.with_context(skip_full_reconcile_check=True)
+        amount_diff = self.company_id.currency_id.round(amount_diff)
+        diff_in_currency = currency.round(diff_in_currency)
+
+        for aml in amls_to_fix:
+            account_expense_line = MoveLineObj.create({
+                'name': _('Currency exchange rate difference'),
+                'debit': amount_diff < 0 and -aml.amount_residual or 0.0,
+                'credit': amount_diff > 0 and aml.amount_residual or 0.0,
+                'account_id': self.debit_move_id.account_id.id,
+                'move_id': move.id,
+                'currency_id': currency.id,
+                'amount_currency': -aml.amount_residual_currency,
+                'partner_id': self.debit_move_id.partner_id.id,
+            })
+
+            account_xchange_gain_line = {
+                'name': _('Currency exchange rate difference'),
+                'debit': amount_diff > 0 and aml.amount_residual or 0.0,
+                'credit': amount_diff < 0 and -aml.amount_residual or 0.0,
+                'account_id': amount_diff > 0 and self.company_id.currency_exchange_journal_id.default_debit_account_id.id or self.company_id.currency_exchange_journal_id.default_credit_account_id.id,
+                'move_id': move.id,
+                'currency_id': currency.id,
+                'amount_currency': aml.amount_residual_currency,
+                'partner_id': self.debit_move_id.partner_id.id}
+
+            MoveLineObj.create(account_xchange_gain_line)
+            partial_rec = (super(AccountPartialReconcile, PartialRecObj)
+                           .create({'debit_move_id': aml.credit and account_expense_line.id or aml.id,
+                                    'credit_move_id': aml.debit and account_expense_line.id or aml.id,
+                                    'amount': abs(aml.amount_residual),
+                                    'amount_currency': abs(aml.amount_residual_currency),
+                                    'currency_id': currency.id,
+                                    }))
+
+            MoveLineObj |= account_expense_line
```

Comment:
Actual recordsets should be underscore_separated, PascalCase is for empty recordsets used as "models". Splitting these two uses here would probably be a good idea.